explo vender stuff

TheDavestDave opened the pull request:
adds a bunch of job gear to the explo vender. Suits, both normal phasers, catalougers, headsets, and now glow sticks, flares, and a new bluespace glow stick. A flare that doesn't run out and burn stuff more or less.

Novacat commented:
I am not really a fan of this PR, mostly for the following reasons.
1: You can already get more holdouts/regular phasers from Cargo.
2: Replacement explorer/pilot supplies should be things acquired from cargo in job cargo crates, much like every other job has.

TheDavestDave commented:
I haven't figured out how crates work, and yes they should be avilable via cargo.

I also feel we should have a way to equipt event charaters we adopt, in the case of phasers it's 40 cargo points for a box with two full sized and two hold outs I think. I'll dig a little and see what I can figure out

TheDavestDave commented:
I've added all the job-none gun things to the new explorer equipment kit. And put the compact catalouger in the vender, instead of the full size one. Removed the phaser from the vender, and added the holdout at 500 points. Puts it in I need a phaser now not I can darg the whole team home to get a real one later kinda thing

Arokha commented on code/modules/mining/ore_redemption_machine/survey_vendor.dm:
```diff
@@ -64,6 +66,11 @@
 		EQUIPMENT("Jump Boots",									/obj/item/clothing/shoes/bhop,										250),
 		EQUIPMENT("Mini-Translocator",							/obj/item/device/perfect_tele/one_beacon,							120),
 		EQUIPMENT("Survival Equipment - Insulated Poncho",		/obj/random/thermalponcho,											75),
+		EQUIPMENT ("Survival Equipment- Glowstick", 			/obj/item/device/flashlight/glowstick,								10),
+		EQUIPMENT ("Survival Equipment- flare", 				/obj/item/device/flashlight/flare,									20),
+		EQUIPMENT ("Survival Equipment- bluespace glowstick", 	/obj/item/device/flashlight/glowstick,								40),
+		EQUIPMENT ("Survival Equipment- explorer headset", 		/obj/item/device/radio/headset/explorer,							40),
+		EQUIPMENT ("Survival Equipment- explorer suit",			/obj/item/clothing/suit/storage/hooded/explorer,					100),
```
```suggestion
		EQUIPMENT ("Survival Equipment - Glowstick", 			/obj/item/device/flashlight/glowstick,								10),
		EQUIPMENT ("Survival Equipment - Flare", 				/obj/item/device/flashlight/flare,									20),
		EQUIPMENT ("Survival Equipment - Bluespace Glowstick", 	/obj/item/device/flashlight/glowstick,								40),
		EQUIPMENT ("Survival Equipment - Explorer Headset", 		/obj/item/device/radio/headset/explorer,							40),
		EQUIPMENT ("Survival Equipment - Explorer Suit",			/obj/item/clothing/suit/storage/hooded/explorer,					100),
```

Doesn't follow the format. Suggested change.

TheDavestDave commented:
I changed the hold out to one that changes faster, I wanted to make a better explo suit but I didn't want to make it better than a void suit and I couldn't figure out how to give it bigger pockets

Novacat commented on code/game/objects/items/devices/flashlight.dm:
```diff
@@ -452,6 +452,15 @@
 	icon_state = "glowstick_yellow"
 	item_state = "glowstick_yellow"
 
+/obj/item/device/flashlight/glowstick/bluespace
```
This will need to go in a _vr file since flashlight.dm is a polaris file.

TheDavestDave replied:
done

Novacat commented on code/datums/supplypacks/supply.dm:
```diff
@@ -143,3 +143,18 @@
+
+/datum/supply_pack/explorergear
+	name="explorer gear"
+	contains = list(
```
Should have a machete sheath, explorer belt, explorer uniform, and explorer headset as well.

TheDavestDave replied:
done

Novacat commented on code/datums/supplypacks/supply.dm:
```diff
@@ -143,3 +143,18 @@
+					/obj/item/device/flashlight/glowstick,
+					/obj/item/stack/marker_beacon/thirty,
+					)
+	cost=10
```
Should cost 20-25, 10 is way too cheap.

Novacat commented on code/modules/clothing/suits/armor_vr.dm:
```diff
@@ -118,4 +118,9 @@
+
+//deluixe explorer suit
+	/obj/item/clothing/suit/armor/pcarrier/explorer/deluxe
+	name = "deluxe explorer suit"
```
rename to "modular explorer suit"

Novacat commented on code/modules/mining/ore_redemption_machine/survey_vendor.dm:
```diff
@@ -42,7 +42,8 @@
 		new /datum/data/mining_equipment("Defense Equipment - Steel Machete",	/obj/item/weapon/material/knife/machete,					50),
-		new /datum/data/mining_equipment("Survival Equipment - Insulated Poncho",	/obj/random/thermalponcho,								75)
+		new /datum/data/mining_equipment("Survival Equipment - Insulated Poncho",	/obj/random/thermalponcho,								75),
+		new /datum/data/mining_equipment("Survey Tools - Compact Catalouger",				/obj/item/device/cataloguer/compact,				50),
```
prize_list is not actually used at all. I am not sure why it exists. Also Compact cataloguers should be in a supplypack, I would recommend making a pilot supply pack alongside the explorer one, since the pilot locker has a few unique things as awell.

TheDavestDave replied:
I don't know what prize_list is, I've just been copy and pasting code and it just works

Novacat commented on code/modules/mining/ore_redemption_machine/survey_vendor.dm:
```diff
@@ -55,6 +56,7 @@
 		EQUIPMENT("Defense Equipment - Razor Drone Deployer",	/obj/item/weapon/grenade/spawnergrenade/manhacks/station/locked,	100),
 		EQUIPMENT("Defense Equipment - Sentry Drone Deployer",	/obj/item/weapon/grenade/spawnergrenade/ward,						150),
 		EQUIPMENT("Defense Equipment - Frontier Carbine",		/obj/item/weapon/gun/energy/locked/frontier/carbine,				750),
+		EQUIPMENT("Defense Equipment - Frontier Holdout Plus",	/obj/item/weapon/gun/energy/locked/frontier/holdout/holdoutplus,	500),
```
I do not feel like there needs to be a Holdout plus, on top of that aside from stats it has no distinction from the regular holdout. This should be removed.

TheDavestDave replied:
it should  charge as fast as the carbine, if I haven't timed it wrong

TheDavestDave replied:
it's gone from the list

Novacat commented on code/modules/mining/ore_redemption_machine/survey_vendor.dm:
```diff
@@ -64,6 +66,11 @@
+		EQUIPMENT ("Survival Equipment - Flare", 				/obj/item/device/flashlight/flare,									20),
+		EQUIPMENT ("Survival Equipment - Bluespace Glowstick", 	/obj/item/device/flashlight/glowstick/bluespace,					40),
+		EQUIPMENT ("Survival Equipment - Explorer Headset", 	/obj/item/device/radio/headset/explorer,							40),
+		EQUIPMENT ("Survival Equipment - Deluxe Explorer Suit",	/obj/item/clothing/suit/armor/pcarrier/explorer/deluxe,				100),
```
The upgraded explorer suit is a bit too cheap.

TheDavestDave replied:
it's at 150 now, you do have to add your own amour

Novacat commented on code/modules/mining/ore_redemption_machine/survey_vendor.dm:
```diff
@@ -64,6 +66,11 @@
+		EQUIPMENT ("Survival Equipment - Glowstick", 			/obj/item/device/flashlight/glowstick,								10),
+		EQUIPMENT ("Survival Equipment - Flare", 				/obj/item/device/flashlight/flare,									20),
+		EQUIPMENT ("Survival Equipment - Bluespace Glowstick", 	/obj/item/device/flashlight/glowstick/bluespace,					40),
+		EQUIPMENT ("Survival Equipment - Explorer Headset", 	/obj/item/device/radio/headset/explorer,							40),
```
This should be in the supply pack at cargo rather than in here.

Novacat reviewed (changes requested):
If you want a recommendation for an upgraded piece of explorer gear to add to the vendor, I would recommend the armored explorer uniform. Already in the code. Price it around 250-300.
/obj/item/clothing/under/explorer/armored
